server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/create/CreateSnapshotRequest.java

```diff
@@ -81,6 +81,7 @@ public class CreateSnapshotRequest extends MasterNodeRequest<CreateSnapshotReque
 
     @Nullable
     private Map<String, Object> userMetadata;
+    private String uuid = null;
```

Comment:
I'm having a bit of trouble with this one. 

It makes sense that `source` and `toXContent` should be consistent, but it's seeming a bit weird to parse `uuid` within source. It seems this would allow users to specify a `uuid` within their SLM policy definition config.

On the flipside, now I'm wondering if `uuid` should be in CreateSnapshotRequest.toXContent? When is a snapshot request even converted to xcontent? I don't ~think~ it's stored in the cluster state or return it via an api endpoint.

Reply:
It appears that `toXContent()` is only used to test `source`. Since `source` shouldn't parse uuid, it could be removed from toXContent. Alternatively, it could just be in toXContent but not in source. This is how the two other fields set in the constructor, `repository` and `snapshot`, are treated. But they could also just be removed from x content, since it's only used for testing. What do you think?

Reply:
Sounds fine to me to remove it from `toXContent`